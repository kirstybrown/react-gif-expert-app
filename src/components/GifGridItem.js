import React from 'react';

const GifGridItem = ( { title, url } ) => {

  return (
        <div className='card animate__animated animate__rubberBand'>
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    );
};

export default GifGridItem;
