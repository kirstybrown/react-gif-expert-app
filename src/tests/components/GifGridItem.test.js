import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';


describe('Tests for GifGridItem component', () => {

    const title = 'elephants';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should show <GifGridItem/> correctly', () => {                
        expect( wrapper ).toMatchSnapshot();        
    });

    test('should show title sent by props in a paragraph', () => {
      const paragraphText = wrapper.find('p').text().trim();
      expect( paragraphText ).toBe(title);
    });

    test('should have an image with url and alt sent by props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
    
    test('should have animate__rubberBand', () => {
       const div = wrapper.find('div');
       const className = div.prop('className');
       expect( className.includes('animate__rubberBand')).toBe(true);
    });
    
    

});
