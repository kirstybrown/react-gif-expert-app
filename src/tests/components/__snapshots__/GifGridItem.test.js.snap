// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGridItem component should show <GifGridItem/> correctly 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="elephants"
    src="https://localhost/something.jpg"
  />
  <p>
     
    elephants
     
  </p>
</div>
`;
